### PR TITLE
feat(flex): simplify use of flex properties

### PR DESCRIPTION
**Pull request by skhamvon**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build#817.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#832.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#833.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#840.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#864.0 ✘  
* 
  * build ✘

File: packages/stencil/components/button/src/components/osds-button/osds-button.scss [hunk 1]
```diff
@@ -45,6 +44,27 @@
   }
 }
 
+
+:host(:not([inline])) {
```
Comment (dpellier) [inline packages/stencil/components/button/src/components/osds-button/osds-button.scss:48]: why not use directly `:host` and remove unnecessary selector ?

File: packages/stencil/components/button/src/index.html [hunk 2]
```diff
@@ -85,8 +85,8 @@
   <osds-button type='submit' size='sm'>submit</osds-button>
 </form>
 
-<p>Flex</p>
-<osds-button flex>Default</osds-button>
+<p>inline</p>
+<osds-button inline=>Default</osds-button>
 
 <p>Ellipsis</p>
 <osds-button flex size='lg'>
```
Comment (dpellier) [inline packages/stencil/components/button/src/index.html:92]: `flex` to remove

File: packages/stencil/components/button/src/components/osds-button/osds-button.scss (hunk 1 above)
Comment (dpellier) [reply packages/stencil/components/button/src/components/osds-button/osds-button.scss:48]: (same question for all of course)

**Review by dpellier — CHANGES_REQUESTED** [review]

**Review by dpellier — COMMENTED** [review]
Please update also stories to replace the `flex` trigger with `inline` one
Capture d’écran 2023-07-11 à 09 06 41

File: packages/stencil/components/chip/src/components/osds-chip/osds-chip.scss [hunk 3]
```diff
@@ -13,15 +13,23 @@
...
+  width: 100%;
 
   @include osds-chip-on-main-element() {
     display: flex;
     align-items: center;
+    width: 100%;
+
+  }
+
+  @include osds-chip-on-main-element() {
```
Comment (dpellier) [inline packages/stencil/components/chip/src/components/osds-chip/osds-chip.scss:30]: duplicate with the one above

**Review by dpellier — CHANGES_REQUESTED** [review]

File: packages/stencil/components/button/src/components/osds-button/osds-button.scss [hunk 4]
```diff
@@ -64,7 +71,7 @@
   pointer-events: none;
 }
 
-:host([flex]) {
+:host(:not([display])) {
```
[the remark this answers is not in the document]
Comment (skhamvon) [reply packages/stencil/components/button/src/components/osds-button/osds-button.scss]: It's done

File: packages/stencil/components/button/src/components/osds-button/osds-button.scss (hunk 1 above)
Comment (skhamvon) [reply packages/stencil/components/button/src/components/osds-button/osds-button.scss]: Done

File: packages/stencil/components/button/src/index.html (hunk 2 above)
Comment (skhamvon) [reply packages/stencil/components/button/src/index.html]: Done

File: packages/stencil/components/chip/src/components/osds-chip/osds-chip.scss (hunk 3 above)
Comment (skhamvon) [reply packages/stencil/components/chip/src/components/osds-chip/osds-chip.scss]: Removed

File: packages/stencil/components/message/src/components/osds-message/osds-message.scss [hunk 5]
```diff
@@ -12,14 +12,16 @@
   align-items: center;
   box-shadow: none;
   box-sizing: border-box;
-  display: inline-flex;
   margin: 0;
   opacity: 1;
   overflow: hidden;
   position: relative;
+  display: flex;
+  width: 100%;
```
[the remark this answers is not in the document]
Comment (skhamvon) [reply packages/stencil/components/message/src/components/osds-message/osds-message.scss]: Removed, but I had to keep the one for range

**Comment by dpellier** [comment]
PR should not be merged, instead changes should be replicated while moving component to the new architecture.
This way we'll avoid multiple breaking versions.
DO NOT DELETE THE BRANCH, as it's referenced in the migration process

**Pull request closed without merging**